utils/vpr_pbtype_arch_wrapper.py

```diff
@@ -0,0 +1,404 @@
+#!/usr/bin/env python3
+"""
+Tool for generate an arch.xml file which includes pb_type.xml and model.xml
+files for testing with Verilog to Routing.
+
+Primarily used by the vpr_test_pbtype cmake function (which is automatically part
+of v2x_test_both cmake function).
+"""
+
+import argparse
+import math
+import os
+import os.path
+import subprocess
+import sys
+import tempfile
+
+from typing import List, Dict, Sequence, Tuple
+
+import lxml.etree as ET
+
+from lib import xmlinc
+from lib.flatten import flatten
+from lib.pb_type import ports, find_leaf, Port
+
+FILEDIR_PATH = os.path.abspath(os.path.join(os.path.dirname(__file__)))
+TEMPLATE_PATH = os.path.abspath(
+    os.path.join(FILEDIR_PATH, "template.arch.xml")
+)
+
+XPos = int
+YPos = int
+GridDict = Dict[Tuple[XPos, YPos], str]
+
+
+def grid_new(width: int, height: int) -> GridDict:
+    """Generate an empty grid dictionary."""
+    tiles = {}
+    for x in range(width):
+        for y in range(0, height):
+            tiles[(x, y)] = '.'
+    return tiles
+
+
+# ABCDEFGHIJKLMNOPQRSTUVWXYZ
+def grid_size(tiles: GridDict) -> Tuple[XPos, YPos]:
+    """Get width and height for a grid dictionary."""
+    width = max(x for x, _ in tiles.keys()) + 1
+    height = max(y for _, y in tiles.keys()) + 1
+    return (width, height)
+
+
+def grid_format(tiles: GridDict) -> str:
```

Comment:
Probably worth making a class for GridDict?  These feel like methods, not really free functions.